### PR TITLE
Analytics: Adds IntersectionObserver functionality to client-side

### PR DESCRIPTION
**Pull request by jonkeller**
Paired with #11724

### CODE REVIEW

File: 3p/iframe-transport-client.js [hunk 1]
```diff
@@ -161,6 +165,26 @@ export class IframeTransportContext {
     this.iframeMessagingClient_./*OK*/sendMessage(
         MessageType.IFRAME_TRANSPORT_RESPONSE,
         /** @type {!JsonObject} */
-        (Object.assign({message: data}, this.baseMessage_)));
+        (Object.assign({message: data},
```
Comment (glevitzky) [inline 3p/iframe-transport-client.js]: Why use Object.assign if you're not using an already existing object? Also, you can avoid the cast here by using dict().

Comment (glevitzky) [inline 3p/iframe-transport-client.js]: Why use Object.assign if you're not using an already existing object? Also, you can avoid the cast here by using dict().

Comment (jonkeller) [reply 3p/iframe-transport-client.js]: Ah, good call. I should have removed the Object.assign() when I changed from baseMessage_ to specifying the object outright.

**Comment by jonkeller** [comment]
Outdated. This work was performed in #12217 and #11724.

**Pull request closed without merging**